test/extended/util/client.go

```diff
@@ -735,15 +712,23 @@ func (c *CLI) Run(command string, flags ...string) *CLI {
 	if c.asPClusterKubeconf && !c.withoutNamespace {
 		FatalErr("you are doing something in ns of pcluster, please use WithoutNamespace and set ns in Args, for example, oc.AsPClusterKubeconf().WithoutNamespace().Run(\"get\").Args(\"pods\", \"-n\", \"pclusterns\").Output()")
 	}
-	if !c.withoutNamespace && c.Namespace() != "" {
-		nc.flags = append(nc.flags, fmt.Sprintf("--namespace=%s", c.Namespace()))
-	}
+
 	// TODO: Temp solution for parallelly ececute our test cases
 	// When https://github.com/kcp-dev/kcp/issues/1689 finished
 	// We could make it simply and just use the --kubeconfig instead of --server.
 	if !c.withoutWorkSpaceServer {
-		nc.flags = append(nc.flags, "--server="+c.currentWs.ServerURL)
+		nc.flags = append(nc.flags, "--server="+c.currentWorkSpace.ServerURL)
 	}
+
+	if !c.withoutNamespace {
+		if c.isKCPCommand() && c.currentWorkSpace.CurrentNameSpace != "" {
+			nc.flags = append(nc.flags, fmt.Sprintf("--namespace=%s", c.currentWorkSpace.CurrentNameSpace))
+		}
+		if !c.isKCPCommand() && c.Namespace() != "" {
+			nc.flags = append(nc.flags, fmt.Sprintf("--namespace=%s", c.Namespace()))
```

Comment:
The KCP commands `kubectl ws`, `kubectl kcp`, `kubectl workspaces` seem to not have the flag --namespace. The right logic seems to be: 

```suggestion
		if <the client is interacting with KCP cluster> && !c.isKCPCommand() && c.currentWorkSpace.CurrentNameSpace != "" {
			nc.flags = append(nc.flags, fmt.Sprintf("--namespace=%s", c.currentWorkSpace.CurrentNameSpace))
		}
		if <the client is interacting with PCluster> && c.Namespace() != "" {
			nc.flags = append(nc.flags, fmt.Sprintf("--namespace=%s", c.Namespace()))
```
I thought quite some time how to make this logic simple with possible solutions, but none is a good solution : (

Reply:
In out auto, both PCluster client and KCP cluster client use the same CLI `c`. This is the root cause which brings many  complexities and some framework bugs we encountered. In `func (c *CLI) Run`, maybe we can check with below order:
```
if !c.withoutKubeconf {
}

// then parse the kubeconfig's current context's server, if it matches "https://.../clusters/root", then we know it is interacting with KCP
// for example, we can set isKCP := true.

// then do below
if isKCP {
        if !c.withoutWorkSpaceServer { ... }
        if !c.withoutNamespace && !c.isKCPCommand() { ... }
}
if isKCP {
        if !c.withoutNamespace && c.Namespace() != "" {...}
}
```

Reply:
Note, do not set isKCP as a flag in the `c` struct. This flag should be always runtime set when `Run` is called. I found asPClusterKubeconf has a similar issue, it is set as a flag in the `c` struct, once it is set true, it is forever true, no place unsets it. @zxiao-redhat @Phaow could you create one separate issue to track this bug?
```
func (c CLI) AsPClusterKubeconf() *CLI {
        c.asPClusterKubeconf = true
```